src/shared/components/ncUserAndGroupPicker/NcUserAndGroupPicker.vue

```diff
@@ -0,0 +1,179 @@
+			label="displayName" :user-select="true"
+			@search="asyncFind" @input="addTransfer">
+			<template #noResult>
+				{{ noResultText }}
+			</template>
+		</NcSelect>
+	</div>
+</template>
+
+<script>
+import { generateOcsUrl } from '@nextcloud/router'
+import { getCurrentUser } from '@nextcloud/auth'
+import axios from '@nextcloud/axios'
+import debounce from 'debounce'
+import { NcSelect } from '@nextcloud/vue'
+import formatting from '../../../shared/mixins/formatting.js'
+import ShareTypes from '../../mixins/shareTypesMixin.js'
+import { showError } from '@nextcloud/dialogs'
+import '@nextcloud/dialogs/style.css'
+
+export default {
+
+	components: {
+		NcSelect,
+	},
+
+	mixins: [ShareTypes, formatting],
+
+	props: {
+		newOwnerUserId: {
+			type: String,
+			default: '',
+		},
+		selectUsers: {
+			type: Boolean,
+			default: true,
+		},
+		selectGroups: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			query: '',
+			value: '',
+			loading: false,
+			minSearchStringLength: 1,
+			maxAutocompleteResults: 20,
+			suggestions: [],
+		}
+	},
+
+	computed: {
+		localValue: {
+			get() {
+				return this.newOwnerUserId
+			},
+			set(v) {
+				console.info('newOwnerUserId set to ', v)
+				this.$emit('update:newOwnerUserId', v)
+			},
+		},
+
+		isValidQuery() {
+			return this.query?.trim() && this.query.length >= this.minSearchStringLength
+		},
+
+		options() {
+			if (this.isValidQuery) {
+				return this.suggestions
+			}
+			return []
+		},
+
+		noResultText() {
+			if (this.loading) {
+				return t('tables', 'Searching …')
+			}
+			return t('tables', 'No elements found.')
+		},
+
+		userId() {
+			return getCurrentUser().uid
+		},
+	},
+
+	methods: {
+		addTransfer(selectedItem) {
+			if (selectedItem) {
+				this.localValue = selectedItem.user
+			} else {
+				this.localValue = ''
+			}
+		},
+
+		getShareTypes() {
+			const types = []
+			if (this.selectUsers) {
+				types.push(this.SHARE_TYPES.SHARE_TYPE_USER)
+			}
+			if (this.selectGroups) {
+				types.push(this.SHARE_TYPES.SHARE_TYPE_GROUP)
+			}
+			return types
+		},
+
+		getShareTypeString(){
+			if (this.selectUsers && !this.selectGroups) {
+				return 'User'
+			}
+			else if (!this.selectUsers && this.selectGroups) {
+				return 'Group'
+			}
+			else {
+				return 'User or group'
+			}
+		},
+
+		async asyncFind(query) {
+			this.query = query.trim()
+			if (this.isValidQuery) {
+				this.loading = true
+				await this.debounceGetSuggestions(query)
+			}
+		},
+
+		async getSuggestions(search) {
+			this.loading = true
+			const shareTypes = this.getShareTypes()
+			let shareTypeQueryString = ''
+			shareTypes.forEach(shareType => {
+				shareTypeQueryString += `&shareTypes[]=${shareType}`
+			})
+			const url = generateOcsUrl('core/autocomplete/get?search={searchQuery}&itemType=%20&itemId=%20{shareTypeQueryString}&limit={limit}', { searchQuery: search, shareTypeQueryString, limit: this.maxAutocompleteResults })
```

Comment:
For some reason, the shareType doesn't work. No matter what I use, it only loads users. @juliushaertl

Reply:
My bad. Wasn't concatenating properly. Fixed now